JWBestPracticeApps/AdvancedPlayer/AdvancedPlayer/ViewController.swift

```diff
@@ -0,0 +1,382 @@
+        progressBar.translatesAutoresizingMaskIntoConstraints = false
+        progressBar.tintColor = .red
+        progressBar.backgroundColor = .lightGray
+        view.addSubview(progressBar)
+        progressBar.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 20).isActive = true
+        progressBar.rightAnchor.constraint(equalTo: view.rightAnchor, constant: -20).isActive = true
+        progressBar.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -20).isActive = true
+        progressBar.heightAnchor.constraint(equalToConstant: 5).isActive = true
+    }
+
+    @objc func progressBarTouchDown(_ sender: UISlider) {
+        progressBarScrubbing = true
+    }
+
+    @objc func progressBarTouchUp(_ sender: UISlider) {
+        player.seek(to: Double(progressBar.value))
+        progressBarScrubbing = false
+    }
+
+    // MARK: - 3. Add a custom play/pause button
+
+    private func addPlayPauseButton(toView view: UIView) {
+        playPauseButton = UIButton(frame: CGRect(x: 0, y: 0, width: 50, height: 50))
+        playPauseButton.addTarget(self, action: #selector(playPauseButtonTap(_:)), for: .touchUpInside)
+        if player.getState() == .playing {
+            playPauseButton.setImage(UIImage(systemName: "pause"), for: .normal)
+        } else {
+            playPauseButton.setImage(UIImage(systemName: "play"), for: .normal)
+        }
+        playPauseButton.translatesAutoresizingMaskIntoConstraints = false
+        playPauseButton.backgroundColor = .white
+        view.addSubview(playPauseButton)
+        playPauseButton.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 25).isActive = true
+        playPauseButton.bottomAnchor.constraint(equalTo: progressBar.topAnchor, constant: -20).isActive = true
+        playPauseButton.widthAnchor.constraint(equalToConstant: 50).isActive = true
+        playPauseButton.heightAnchor.constraint(equalToConstant: 50).isActive = true
+    }
+
+    @objc func playPauseButtonTap(_ button: UIButton) {
+        let state = player.getState()
+        if state == .playing {
+            player.pause()
+        } else {
+            player.play()
+        }
+    }
+
+    override func jwplayer(_ player: JWPlayer, didPauseWithReason reason: JWPauseReason) {
+        super.jwplayer(player, didPauseWithReason: reason)
+        DispatchQueue.main.async { [weak self] in
+            self?.playPauseButton.setImage(UIImage(systemName: "play"), for: .normal)
+        }
+    }
+
+    override func jwplayer(_ player: JWPlayer, isPlayingWithReason reason: JWPlayReason) {
+        super.jwplayer(player, isPlayingWithReason: reason)
+        DispatchQueue.main.async { [weak self] in
+            self?.playPauseButton.setImage(UIImage(systemName: "pause"), for: .normal)
+        }
+    }
+
+    // MARK: - 4. Add a custom skip button
+
+    private func addSkipButton(toView view: UIView) {
+        skipButton = UIButton(frame: CGRect(x: 0, y: 0, width: 50, height: 50))
```

Comment:
I feel like these are very verbose which would be expected from setting them programmatically, should we take advantage of a .xib for the view that will be added though the addSubview? I think we can do the same with `JWFullscreenViewController`. And only leave the addition of the skip button here. What do you think?

Reply:
@DavidPerezP124 sounds good to me